src/codec.rs

```diff
@@ -50,21 +68,51 @@ pub trait Decode<D>: Sized {
     }
 }
 
+// Provide a blanket implementation so that any Decode can be used as a ParameterizedDecode<()>.
+impl<D: Decode> ParameterizedDecode<()> for D {
+    fn decode_with_param(
+        _decoding_parameter: &(),
+        bytes: &mut Cursor<&[u8]>,
+    ) -> Result<Self, CodecError> {
+        Self::decode(bytes)
+    }
+}
+
 /// Describes how to encode objects into a byte sequence.
-pub trait Encode {
+pub trait Encode: Sized {
```

Comment:
Why is `Sized` needed? Neither method o n the trait takes `self` by value or deals in `Self`.

Reply:
Without it, compilation fails with the following error:
```
error[E0599]: the method `get_encoded_with_param` exists for reference `&Self`, but its trait bounds were not satisfied
  --> src/codec.rs:88:14
   |
88 |         self.get_encoded_with_param(&())
   |              ^^^^^^^^^^^^^^^^^^^^^^ method cannot be called on `&Self` due to unsatisfied trait bounds
   |
   = note: the following trait bounds were not satisfied:
           `Self: Sized`
           which is required by `Self: ParameterizedEncode<()>`
           `&Self: Encode`
           which is required by `&Self: ParameterizedEncode<()>`
```


I think this was because of the implicit `Sized` bound in the blanket implementation; I added an explicit `?Sized` and compilation works out without specifying that `Encode` requires `Sized`.

Reply:
In any case I don't think requiring `Sized` is a problem here.